name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/AdvanziaBankPDFExtractor.java

```diff
@@ -0,0 +1,113 @@
+                        // @formatter:on
+                                        .section("currency")
+                                        .match("^Datum Beschreibung Betrag \\((?<currency>[\\w]{3})\\).*$")
+                                        .assign((ctx, v) -> ctx.put("currency", asCurrencyCode(v.get("currency")))));
+
+        this.addDocumentTyp(type);
+
+        // @formatter:off
+        // 30.06.2023 ZINSERTRAG 47,11
+        // @formatter:on
+        Block interestBlock = new Block("^[\\d]{2}\\.[\\d]{2}\\.[\\d]{4} ZINSERTRAG [\\.\\d]+,[\\d]{2}.*$");
+        type.addBlock(interestBlock);
+        interestBlock.set(new Transaction<AccountTransaction>()
+
+                        .subject(() -> {
+                            AccountTransaction accountTransaction = new AccountTransaction();
+                            accountTransaction.setType(AccountTransaction.Type.INTEREST);
+                            return accountTransaction;
+                        })
+
+                        .section("date", "amount") //
+                        .documentContext("currency") //
+                        .match("^(?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) ZINSERTRAG (?<amount>[\\.\\d]+,[\\d]{2}).*$") //
+                        .assign((t, v) -> {
+                            t.setDateTime(asDate(v.get("date")));
+                            t.setAmount(asAmount(v.get("amount")));
+                            t.setCurrencyCode(v.get("currency"));
+                        })
+
+                        .wrap(TransactionItem::new));
+
+        // @formatter:off
+        // 14.06.2023 EINZAHLUNG 100,00
+        // @formatter:on
+        Block depositBlock = new Block("^[\\d]{2}\\.[\\d]{2}\\.[\\d]{4} EINZAHLUNG [\\.\\d]+,[\\d]{2}.*$");
+        type.addBlock(depositBlock);
+        depositBlock.set(new Transaction<AccountTransaction>()
+
+                        .subject(() -> {
+                            AccountTransaction accountTransaction = new AccountTransaction();
+                            accountTransaction.setType(AccountTransaction.Type.DEPOSIT);
+                            return accountTransaction;
+                        })
+
+                        .section("date", "amount") //
+                        .documentContext("currency") //
+                        .match("^(?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) EINZAHLUNG (?<amount>[\\.\\d]+,[\\d]{2}).*$") //
+                        .assign((t, v) -> {
+                            t.setDateTime(asDate(v.get("date")));
+                            t.setAmount(asAmount(v.get("amount")));
+                            t.setCurrencyCode(v.get("currency"));
+                        })
+
+                        .wrap(TransactionItem::new));
+
+        // @formatter:off
+        // 15.06.2023 AUSZAHLUNG -100,00
+        // @formatter:on
+        Block removalBlock = new Block("^[\\d]{2}\\.[\\d]{2}\\.[\\d]{4} AUSZAHLUNG -[\\.\\d]+,[\\d]{2}.*$");
```

Comment:
```suggestion
        Block removalBlock = new Block("^[\\d]{2}\\.[\\d]{2}\\.[\\d]{4} AUSZAHLUNG \\-[\\.\\d]+,[\\d]{2}.*$");
```

Reply:
AFAIK, a hyphen outside a character class is treated as a regular character by the regex. So it should be fine without escaping it.

Should I still change it to be alinged with other other PDF importers?
(I compared mine to Santander Bank)

Reply:
> Should I still change it to be alinged with other other PDF importers?
> (I compared mine to Santander Bank)

Yes, that would be great.